### PR TITLE
implement ping endpoint

### PR DESCRIPTION
**Pull request by GiancarloFusiello**
relates to issue [109](https://github.com/libero/libero/issues/109)

### CODE REVIEW

File: search/ping.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from flask import Blueprint, Response
+
+
+def get_ping_blueprint() -> Blueprint:
+    blueprint = Blueprint('ping', __name__)
+
+    @blueprint.route('/ping', methods=['GET'])
+    def search() -> Response:  # pylint: disable=unused-variable
```
Comment (thewilkybarkid) [inline search/ping.py]: Name needs updating

File: search/ping.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        response = Response(response='pong', status=200, mimetype='text/plain')
+        response.headers['Cache-Control'] = 'no-store, must-revalidate'
```
Comment (thewilkybarkid) [inline search/ping.py:10]: Need to also follow `It MUST return a Expires header with the value 0 if HTTP/1.0 clients/proxies are used`

**Comment by thewilkybarkid** [comment]
Since this route is to be replicated in all services, it's the ideal candidate for the first Python library ([we did this with PHP too](https://github.com/libero/ping-controller)).

File: search/ping.py (hunk 2 above)
Comment (GiancarloFusiello) [reply search/ping.py:10]: updated.

File: search/ping.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+from flask import Blueprint, Response, request
+
+
+def get_ping_blueprint() -> Blueprint:
+    blueprint = Blueprint('ping', __name__)
+
+    @blueprint.route('/ping', methods=['GET'])
+    def ping() -> Response:  # pylint: disable=unused-variable
```
Comment (giorgiosironi) [inline search/ping.py:8]: out of curiosity what's the unused variable here?

Comment (GiancarloFusiello) [reply search/ping.py:8]: Pylint thinks that the function declaration is a variable and that it's not being used. It appears to be an known issue with Pylint and Flask decorators.

Comment (thewilkybarkid) [reply search/ping.py:8]: Related: https://github.com/elifesciences/profiles/pull/14#discussion_r138336222

File: search/ping.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        response = Response(response='pong', status=200, mimetype='text/plain')
+        response.headers['Cache-Control'] = 'no-store, must-revalidate'
+        if request.environ.get('SERVER_PROTOCOL') == 'HTTP/1.0':
+            response.headers['Expires'] = 0
```
Comment (thewilkybarkid) [inline search/ping.py:12]: Needs testing.

Comment (GiancarloFusiello) [reply search/ping.py:12]: tests added.

**Review by thewilkybarkid — APPROVED** [review]
We should open an issue to extract this into a library.

**Pull request merged**